portfolio/src/main/java/com/google/sps/servlets/MarkerServlet.java

```diff
@@ -57,12 +71,24 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
       String crimeAddress = Jsoup.clean(request.getParameter("address"), Whitelist.none());
       String crimeDescription = Jsoup.clean(request.getParameter("description"), Whitelist.none());
 
-      Marker marker =
-          new Marker(lat, lng, crime, crimeDate, crimeTime, crimeAddress, crimeDescription);
-      storeMarker(marker);
+      if (repeatMarkers(lat, lng, crimeDate, crimeTime, crime)) {
+        resultEnum.setStatus(StoreStatus.FAILURE);
+        resultEnum.setFailureType(StoreFailureType.REPEAT);
+      }
+      else {
+        resultEnum.setStatus(StoreStatus.SUCCESS);
+        Marker marker =
+            new Marker(lat, lng, crime, crimeDate, crimeTime, crimeAddress, crimeDescription);
+        storeMarker(marker);
+      }
+
     } catch (Error error) {
+        resultEnum.setStatus(StoreStatus.FAILURE);
+        resultEnum.setFailureType(StoreFailureType.UNKNOWN);
       System.out.println(error);
     }
+    String statusEnum = gson.toJson(resultEnum);
```

Comment:
nit: no need for this variable, since it is only used once. just directly call the method on line below

Reply:
Done!